geometric_kernels/spaces/lie_groups.py

```diff
@@ -84,25 +87,25 @@ def _addition_theorem(self, X: B.Numeric, X2: B.Numeric, **parameters) -> B.Nume
         diff = self._difference(X, X2)
         torus_repr_diff = self._torus_representative(diff)
         values = [
-            degree**2 * chi(torus_repr_diff)[..., None]  # [N1, N2, 1]
-            for chi, degree in zip(self._characters, self._dimensions)
+            repr_dim * B.real(chi(torus_repr_diff)[..., None])  # [a, b, 1]
+            for chi, repr_dim in zip(self._characters, self._dimensions)
         ]
-        return B.concat(*values, axis=-1)  # [N, N2, L]
+        return B.concat(*values, axis=-1)  # [a, b, L]
 
     def _addition_theorem_diag(self, X: B.Numeric, **parameters) -> B.Numeric:
         """
         Returns the sum of eigenfunctions on a level for which we have a simplified expression
-        :param X: [N, D]
+        :param X: [a, n, n]
```

Comment:
I think the convention is to use "N" (and similar) for the number of points and "D" or "D, D" for dimensions.

Reply:
Done.